BridgeCareApp/BridgeCare/Services/SummaryReport/SummaryReportBridgeData.cs

```diff
@@ -0,0 +1,361 @@
+            var bridgeDataModels = bridgeData.GetBridgeData(BRKeys, dbContext);
+                        
+            var simulationDataModels = GetSimulationDataModels(simulationDataTable, simulationYears, projectCostModels);
+
+            // Fiil up the excel.
+            var headers = GetHeaders();
+            var currentCell = AddHeadersCells(worksheet, headers, simulationYears);
+
+            // Add row next to headers for fitlers and year no.s for dynamic data. Cover from top, left to right bottom whole set of data
+            using (ExcelRange autoFilterCells = worksheet.Cells[3, 1, currentCell.Row, currentCell.Coln])
+            {
+                autoFilterCells.AutoFilter = true;
+            }
+
+            AddBridgeDataModelsCells(worksheet, bridgeDataModels, currentCell);
+            AddDynamicDataCells(worksheet, sectionsForSummaryReport, simulationDataModels, bridgeDataModels, currentCell);
+            // TODO uncomment later: Postman getting hang if this is kept 
+            // ApplyBorder(worksheet.Cells[1, 1, currentCell.Row, currentCell.Coln]);
+            worksheet.Cells.AutoFitColumns();
+        }
+
+        #region Private Methods
+        private void AddDynamicDataCells(ExcelWorksheet worksheet, List<Section> sectionsForSummaryReport, List<SimulationDataModel> simulationDataModels, List<BridgeDataModel> bridgeDataModels, CurrentCell currentCell)
+        {            
+            var row = 4; // Data starts here
+            var coln = currentCell.Coln;
+            foreach (var bridgeDataModel in bridgeDataModels)
+            {
+                coln = currentCell.Coln;
+                var brKey = bridgeDataModel.BRKey;
+                var familyId = bridgeDataModel.BridgeFamily;
+                var workDoneMoreThanOnce = 0;
+                var section = sectionsForSummaryReport.Where(s => Convert.ToInt32(s.FACILITY) == brKey).FirstOrDefault();
+                var simulationDataModel = simulationDataModels.Where(s => s.SectionId == section.SECTIONID).FirstOrDefault();
+                var yearsData = simulationDataModel.YearsData;
+                // Add work done cellls
+                for (var cnt = 1; cnt < yearsData.Count(); cnt++)
+                {
+                    var cost = yearsData[cnt].Cost;
+                    worksheet.Cells[row, ++coln].Value = cost > 0 ? "Yes" : "--";
+                    workDoneMoreThanOnce = cost > 0 ? workDoneMoreThanOnce + 1 : workDoneMoreThanOnce;
+                }
+                worksheet.Cells[row, ++coln].Value = workDoneMoreThanOnce > 1 ? "Yes" : "--";
+
+                // Empty Total column
+                coln++;
+
+                // Add Poor On/Off Rate column: Formula (prev yr SD == "Y")?(curr yr SD=="N")?"Off":"On":"--"
+                // TODO Ask if prev yr formula is wrong in report, not matching with other yrs formulae.
+                for (var cnt = 1; cnt < yearsData.Count(); cnt++)
```

Comment:
In general, don't use abbreviations in your code.

Reply:
Done.